tracing-journald/src/socket.rs

```diff
@@ -0,0 +1,59 @@
+//! socket helpers.
+
+use std::io::{Error, Result};
+use std::mem::size_of;
+use std::os::unix::net::UnixDatagram;
+use std::os::unix::prelude::{AsRawFd, RawFd};
+use std::ptr;
+
+use libc::*;
+
+const CMSG_BUFSIZE: usize = 64;
+
+#[repr(C)]
+union AlignedBuffer<T: Copy + Clone> {
+    buffer: T,
+    align: cmsghdr,
+}
+
+fn cmsg_space(s: u32) -> usize {
+    unsafe { CMSG_SPACE(s) as usize }
+}
+
+pub fn send_one_fd(socket: &UnixDatagram, fd: RawFd) -> Result<usize> {
+    assert!(
+        cmsg_space(size_of::<RawFd>() as u32) <= CMSG_BUFSIZE,
+        "cmsghdr buffer too small (< {}) to hold a single fd",
+        cmsg_space(size_of::<RawFd>() as u32)
+    );
```

Comment:
nit/TIOLI: it might be worth sticking this assertion in a unit test as well, so that we can get a CI failure if it is ever somehow violated?

Reply:
Shouldn't it already be exercised by the large payload unit test?

Reply:
ah, yeah, you're right, never mind!

Reply:
It's covered by the payload test, but I've still added a separate unit test now, because I think it's easier to debug that way, and it doesn't really add much.